src/main/groovy/grails/melody/plugin/GrailsMelodyPluginGrailsPlugin.groovy

```diff
@@ -40,20 +37,12 @@ class GrailsMelodyPluginGrailsPlugin extends Plugin {
         new MelodyInterceptorEnhancer().enhance(getGrailsApplication())
     }
 
-    void doWithApplicationContext() {
-        //Need to wrap the datasources here, because BeanPostProcessor didn't worked.
-        def beans = getApplicationContext().getBeansOfType(DataSource)
-        beans.each { beanName, bean ->
-            if (bean?.hasProperty('targetDataSource')) {
-                LOG.debug "Wrapping DataSource - $beanName"
-                bean.targetDataSource = JdbcWrapper.SINGLETON.createDataSourceProxy(bean.targetDataSource)
-            }
-        }
-    }
-
     Closure doWithSpring() {
-        {->
+        { ->
             melodyConfig(MelodyConfig)
+
+            //Wrap grails datasource with java melody JdbcWapper
+            grailsDataSourceBeanPostProcessor(GrailsDataSourceBeanPostProcessor)
```

Comment:
I think that GrailsDataSourceBeanPostProcessor is not necessary here.
`net.bull.javamelody.SpringDataSourceBeanPostProcessor` works certainly as well and it will also display info about the datasource in the monitoring report.

Reply:
Ok, I'll try it